packages/generator/src/sdk-metadata/pregenerated-lib.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { resolve } from 'path';
+import { VdmServiceMetadata } from '../vdm-types';
+import { parseService } from '../service-generator';
+import { createOptions } from '../../test/test-util/create-generator-options';
+import { GlobalNameFormatter } from '../global-name-formatter';
+import { oDataServiceSpecs } from '../../../../test-resources/odata-service-specs';
+import { GeneratorOptions } from '../generator-options';
+import {
+  getInstallationSnippet,
+  getRepositoryLink,
+  getVersion,
+  getServiceDescription,
+  getTimeStamp, isPublishedNpmPackage
+} from './pregenerated-lib';
+
+describe('sdk metadata - pregenerated client information',()=>{
+  const service: VdmServiceMetadata = getTestService();
+
+  it('returns installation snipped',()=>{
+    expect(getInstallationSnippet(service)).toBe('npm i @sap/cloud-sdk-vdm-test-service:latest');
+  });
+
+  it('returns repository link',()=>{
+    expect(getRepositoryLink(service)).toBe('https://www.npmjs.com/package/@sap/cloud-sdk-vdm-test-service');
+  });
+
+  it('returns version from generator options or generator version',()=>{
+    expect(getVersion({ versionInPackageJson:'123' } as GeneratorOptions)).toBe('123');
+    expect(getVersion({} as GeneratorOptions)).toMatch(/\d+\.\d+\.\d+/);
+  });
+
+  it('returns description of the service',()=>{
+    expect(getServiceDescription(service,createOptions())).toMatchSnapshot();
+  });
+
+  it('returns a timestamp in unix format',()=>{
+    expect(getTimeStamp()).toMatch(/\/Date\(\d{13,13}\)\//);
+  });
+
+  it('checks if there is a client published',async ()=>{
```

Comment:
Could we mark these newly added e2e tests with e.g. tags [E2E] so once we want to take further actions, we can easily search for them.